### ISSUE TITLE
Add labels for visible points

### ISSUE THREAD

**Issue by crazydef** [awulkiew/graphical-debugging#34]
In the Geometry Watch there are no labels giving any indication as to which point is which (from a vector of points).

Please add either labels (that appear or disappear according to the zoom level and the data they would otherwise obstruct) or add tooltips when hovering over a point.

**Comment by awulkiew** [comment]
Right, good suggestion. In case of array or `vector`/`List` an obvious choice for a label is the index of a point in the container. However in cases such as `list`/`LinkedList` or `set`/`map`/`Dictionary` it's not clear what it should be. An index from the beginning (whatever it is) is a straightforward choice but maybe do you have another idea?

**Comment by crazydef** [comment]
A permanently displayed index would have been sufficient for my needs.

Lists should just display the index too.  In the case of a map, though, there is no implied relationship between the points, so none should be stated.

As far as tooltips are concerned, it would be nice to see exact values when hovering over a point.  And even relative values if it's possible to click on two or more points to see their relation to each other.  (Specifically, it would have been useful for me to be able to see relationships between points from two different vectors/polygons.)
